docs/src/components/home/CoreShowcase.tsx

```diff
@@ -0,0 +1,343 @@
+          },
+        },
+      },
+    },
+  },
+});
+
+const CODES = [
+  `const theme = createTheme({
+  shape: {
+    borderRadius: 10,
+  },`,
+  `
+  spacing: 10,`,
+  `
+  shadows: [
+    'none',
+    '0px 4px 20px 0px hsla(210, 14%, 28%, 0.2)',
+  ],`,
+  `
+  typography: {
+    fontFamily: '"PlusJakartaSans", sans-serif',
+    fontWeightBold: 500,
+  },`,
+  `
+  palette: {
+    background: {
+      default: theme.palette.mode === 'dark' ? '#0A1929' : '#F3F6F9',
+    },`,
+  `
+    divider: '#E5E8EC',`,
+  `
+    primary: {
+      main: '#007FFF',
+    },`,
+  `
+    text: {
+      primary: '#3D4752',
+      secondary: '#5A6978',
+    },`,
+  `
+    success: {
+      main: '#1AA251',
+      light: '#6AE79C',
+    },
+  },`,
+  `
+  components: {
+    MuiAvatar: {
+      styleOverrides: {
+        root: {
+          width: 64,
+          height: 64,
+        },
+      },
+    },`,
+  `
+    MuiCard: {
+      styleOverrides: {
+        root: {
+          border: '1px solid',
+          borderColor: '#D7DCE1',
+        },
+      },
+    },`,
+  `
+    MuiIconButton: {
+      styleOverrides: {
+        root: {
+          border: '1px solid',
+          borderColor: '#E5E8EC',
+          color: '#5A6978',
+          '&:hover, &.Mui-focusVisible': {
+            borderColor: '#007FFF',
+            color: '#007FFF',
+          },
+        },
+      },
+    },`,
+  `
+    MuiSwitch: {
+      styleOverrides: {
+        root: { width: 32, height: 20, padding: 0 },
+        switchBase: { height: 20, width: 20, padding: 0,
+          '&.Mui-checked + .MuiSwitch-track': {
+            opacity: 1,
+          },
+          '&.Mui-checked': {
+            transform: 'translateX(11px)',
+            color: '#fff',
+          },
+        },
+        track: {
+          opacity: 1,
+          borderRadius: 32,
+          backgroundColor: '#BFC7CF'
+        },
+        thumb: { width: 14, height: 14 },
+      },
+    },
+  },
+});`,
+];
+
+const noop = () => {};
+
+function TypeWriter({
+  step = 0,
+  codes = CODES,
+  onStepComplete = noop,
+}: {
+  step: number;
+  codes: Array<string>;
+  onStepComplete: (step: number) => void;
+}) {
+  const [result, setResult] = React.useState(``);
+  const index = React.useRef(0);
+  React.useEffect(() => {
+    index.current = 0;
+    if (step > 0) {
+      // by line
+      // const code = codes[step - 1].split('\n').map((text, i, array) => {
+      //   if (i === array.length - 1) return text;
+      //   return `${text}\n`;
+      // });
+
+      // by character
+      let code = codes[step - 1].split('');
+      code = code.reduce((final, curr) => {
+        if (
+          curr.match(/^\s*$/) &&
+          final[final.length - 1] &&
+          final[final.length - 1].match(/^\s*$/)
+        ) {
+          return [...final.slice(0, final.length - 1), `${final[final.length - 1]} `];
+        }
+        return [...final, curr];
+      }, [] as Array<string>);
+
+      const time = setInterval(() => {
```

Comment:
Not sure if it is exactly here but I feel like we can make it just a bit faster 😅

Reply:
set to 32ms/letter (before was 50ms)